src/main/java/com/adyen/model/checkout/PaymentsRequest.java

```diff
@@ -593,41 +581,29 @@ public void setOrderReference(String orderReference) {
         this.orderReference = orderReference;
     }
 
-    public PaymentsRequest paymentMethod(Map<String, String> paymentMethod) {
-        this.paymentMethod = paymentMethod;
-        return this;
-    }
-
-    public PaymentsRequest putPaymentMethodItem(String key, String paymentMethodItem) {
-
-        this.paymentMethod.put(key, paymentMethodItem);
-        return this;
+    public PaymentMethodDetails getPaymentMethod() {
+        return paymentMethod;
     }
 
-    /**
-     * The collection that contains the type of the payment method and its specific information (e.g. &#x60;idealIssuer&#x60;).
-     *
-     * @return paymentMethod
-     **/
-    public Map<String, String> getPaymentMethod() {
-        return paymentMethod;
+    public void setPaymentMethod(PaymentMethodDetails paymentMethod) {
+        this.paymentMethod = paymentMethod;
     }
 
-    public void setPaymentMethod(Map<String, String> paymentMethod) {
+    public PaymentsRequest paymentMethod(PaymentMethodDetails paymentMethod) {
         this.paymentMethod = paymentMethod;
+        return this;
     }
 
     public PaymentsRequest addEncryptedCardData(String encryptedCardNumber, String encryptedExpiryMonth, String encryptedExpiryYear, String encryptedSecurityCode, String holderName) {
-        this.paymentMethod = new HashMap<>();
-        this.paymentMethod.put(METHOD_TYPE, TYPE_SCHEME);
-        this.paymentMethod.put(ENCRYPTED_CARD_NUMBER, encryptedCardNumber);
-        this.paymentMethod.put(ENCRYPTED_EXPIRY_MONTH, encryptedExpiryMonth);
-        this.paymentMethod.put(ENCRYPTED_EXPIRY_YEAR, encryptedExpiryYear);
+        DefaultPaymentMethodDetails paymentMethodDetails = new DefaultPaymentMethodDetails();
+        this.paymentMethod = paymentMethodDetails;
```

Comment:
minor: exchange of 599 and 601 will make it more readable.

Reply:
done